src/schematic/client.py

```diff
@@ -275,10 +306,36 @@ async def _enqueue_event(self, event_type: str, body: EventBody) -> None:
     def _get_flag_default(self, flag_key: str) -> bool:
         return self.flag_defaults.get(flag_key, False)
 
-    def _shutdown_handler(self, signum, frame):
-        self.logger.info(f"Received signal {signum}. Initiating shutdown.")
-        asyncio.create_task(self.shutdown())
-
     async def shutdown(self) -> None:
-        await self.event_buffer.stop()
-        self.logger.info("Shutdown complete.")
+        """Properly shutdown the client, flushing any pending events.
+        
+        This method should be called when you're done using the client to ensure:
+        - All pending events are flushed to the server
+        - Background tasks are properly terminated
+        - Resources are cleaned up
+        
+        It's safe to call this method multiple times, even if the client was never used.
+        """
+        # Only do the shutdown once
+        if self._is_shutting_down:
+            self.logger.debug("Shutdown already in progress, skipping")
+            return
+            
+        self._is_shutting_down = True
+        
+        # If we were never initialized, there's nothing to clean up
+        if not self._initialized:
```

Comment:
self._initialized is set to true at the end of __init__

is this meant to guard against a shut down happening during initialization? if so, should we set _initialized at the beginning of that function?